src/trifinger_platform_log.cpp

```diff
@@ -0,0 +1,91 @@
+/**
+ * @file
+ * @brief Access combined log of a TriFinger platform.
+ * @copyright 2020, Max Planck Gesellschaft.  All rights reserved.
+ * @license BSD 3-clause
+ */
+#include <robot_fingers/trifinger_platform_log.hpp>
+
+namespace robot_fingers
+{
+TriFingerPlatformLog::TriFingerPlatformLog(const std::string &robot_log_file,
+                                           const std::string &camera_log_file)
+    : robot_log_(robot_log_file),
+      camera_log_(camera_log_file),
+      map_robot_to_camera_index_(robot_log_.data.size(), -1)
+{
+    if (robot_log_.data.empty())
+    {
+        throw std::runtime_error("Robot log is empty");
+    }
+    robot_log_start_index_ = robot_log_.data.front().timeindex;
+
+    // for each robot log entry, find the matching camera log entry
+    int i_camera = 0;
+    for (auto i_robot = 0; i_robot < robot_log_.data.size(); i_robot++)
+    {
+        // timestamp in robot log is in seconds, convert to milliseconds
+        auto stamp_robot_ms = robot_log_.data[i_robot].timestamp * 1000;
+
+        while (i_camera < camera_log_.timestamps.size() &&
+               stamp_robot_ms >= camera_log_.timestamps[i_camera])
+        {
+            i_camera++;
+        }
+        map_robot_to_camera_index_[i_robot] = i_camera - 1;
+    }
+}
+
+TriFingerPlatformLog::RobotObservation
+TriFingerPlatformLog::get_robot_observation(const time_series::Index &t) const
+{
+    return robot_log_.data.at(t - robot_log_start_index_).observation;
```

Comment:
I think you assume throughout the class that no timeindices are missing, i.e. that

robot_log_.data[i+1].timeindex =  robot_log_.data[i].timeindex + 1.

is this safe to assume? if you think it makes sense, maybe you could check this in the constructor.

Reply:
I think it is safe to assume but it might still be good to add some check to be sure.

Reply:
Done